#166731421 Feature pagination support

despeauxz opened the pull request:
#### What does this PR do?
This PR adds Pagination support to the entire application
#### Description of Task to be completed?
Check the `GET ratings route`
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#166731421
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

despeauxz committed:
feat-[166731421]: pagination support

- pagination helper for bulky responses
- paginate the get ratings page

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: pagination support

- pagination helper for bulky responses
- paginate the get ratings page

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: pagination support

- pagination helper for bulky responses
- paginate the get ratings page

[Finishes #166731421]

despeauxz committed:
Merge branch '166731421-ft-pagination-support' of https://github.com/andela/kingsmen-ah-backend into 166731421-ft-pagination-support

despeauxz committed:
feat-[166731421]: pagination support

- pagination helper for bulky responses
- paginate the get ratings page

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: pagination support

- pagination helper for bulky responses
- paginate the get ratings page

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: pagination support

- pagination helper for bulky responses
- paginate the get ratings page

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: pagination support

- pagination helper for bulky responses
- paginate the get ratings page

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: pagination support

- pagination helper for bulky responses
- paginate the get ratings page

[Finishes #166731421]

despeauxz committed:
Merge branch '166731421-ft-pagination-support' of https://github.com/andela/kingsmen-ah-backend into 166731421-ft-pagination-support

despeauxz committed:
Merge branch '166731421-ft-pagination-support' of https://github.com/andela/kingsmen-ah-backend into 166731421-ft-pagination-support

fidelisojeah commented on server/controllers/articles.js:
```diff
@@ -214,6 +219,58 @@ class ArticleController {
+
+      const { id: articleId } = article.dataValues;
+
+      const ratings = await Rating.findAndCountAll({
```
unfortunately, the `rating count` you are using here can't fly or make much sense.

if you want a count, you need to run a separate count query without the `limit` and `offset` but should include the `where clause`

despeauxz replied:
I'm on it boss

despeauxz committed:
Merge branch '166731421-ft-pagination-support' of https://github.com/andela/kingsmen-ah-backend into 166731421-ft-pagination-support

despeauxz committed:
Merge branch '166731421-ft-pagination-support' of https://github.com/andela/kingsmen-ah-backend into 166731421-ft-pagination-support

despeauxz committed:
Merge branch '166731421-ft-pagination-support' of https://github.com/andela/kingsmen-ah-backend into 166731421-ft-pagination-support

despeauxz committed:
Merge branch 'develop' of https://github.com/andela/kingsmen-ah-backend into 166731421-ft-pagination-support

despeauxz committed:
feat-[166731421]: Pagination support

- paginate the get all ratings endpoint
- paginate the get all articles endpoint
- test the endpoints

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: Pagination support

- paginate the get all ratings endpoint
- paginate the get all articles endpoint
- test the endpoints

[Finishes #166731421]

fidelisojeah commented on server/controllers/articles.js:
```diff
@@ -214,6 +221,59 @@ class ArticleController {
+      if (!article) return Response.error(res, 404, 'Article does not exist');
+
+      const { id: articleId } = article.dataValues;
+      const ratingsDetails = await ratingsPayload(articleId);
```
this is totally not necessary

you only need to count

`const count = Rating.count({where: {articleId}});`

and then get ratings
```javascript
const ratings = Rating.findAll({ 
  where: { articleId },
  limit,
  offset,
...
```

you should use same step for article

despeauxz replied:
Thanks for the feedback, was really helpful

fidelisojeah commented on server/helpers/articlePayload.js:
```diff
@@ -1,12 +1,60 @@
+  });
+};
+
+const findArticleCount = async () => Article.findAll({
```
unnecessary.

fidelisojeah commented on server/helpers/ratingPayload.js:
```diff
@@ -0,0 +1,25 @@
+
+const { Rating, User, Profile } = models;
+
+const findRatings = articleId => Rating.findAll({
```
unnecessary

despeauxz committed:
feat-[166731421]: Pagination support

- paginate the get all ratings endpoint
- paginate the get all articles endpoint
- test the endpoints

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: Pagination support

- paginate the get all ratings endpoint
- paginate the get all articles endpoint
- test the endpoints

[Finishes #166731421]

despeauxz committed:
feat-[166731421]: Pagination support

- paginate the get all ratings endpoint
- paginate the get all articles endpoint
- test the endpoints

[Finishes #166731421]

fidelisojeah commented on server/routes/api/articles.js:
```diff
@@ -20,4 +20,7 @@ articlesRouter.post('/', Token.authorize, ArticleController.create);
 articlesRouter.delete('/:slug', Token.authorize, ArticleController.delete);
 
+articlesRouter.get('/:slug/rate', ArticleController.getArticleRatings);
+articlesRouter.post('/:slug/rate', trim, Token.authorize, ArticleController.rate);
```
duplicate
```suggestion

```

despeauxz replied:
I'll fix it right now

despeauxz committed:
feat-[166731421]: Pagination support

- paginate the get all ratings endpoint
- paginate the get all articles endpoint
- test the endpoints

[Finishes #166731421]

adex001 reviewed (approved):
LGTM. Good work!
